code/main/task/eventProcessor.brs

```diff
@@ -29,11 +29,12 @@ function _adb_EventProcessor(task as object) as object
         init: function() as void
             m._configurationModule = _adb_ConfigurationModule()
             m._consentState = _adb_ConsentState(m._configurationModule)
-            m._identityModule = _adb_IdentityModule(m._configurationModule, m._consentState)
-            m._edgeModule = _adb_EdgeModule(m._configurationModule, m._identityModule, m._consentState)
+            m._identityState = _adb_IdentityState()
+            m._edgeModule = _adb_EdgeModule(m._configurationModule, m._identityState, m._consentState)
+            m._identityModule = _adb_IdentityModule(m._identityState, m._edgeModule, m._task)
```

Comment:
Instead of passing the task to the identity module, it would be cleaner to modify the getECID function to accept a callback. The function can then notify the task through the callback, so other modules don’t need to be aware of the task.

Reply:
@praveek sounds good. Updated to callbacks!

Reply:
@praveek Also fixed issues like:

```
- All the waiting callbacks will be called
- Callback map will be cleared once all callbacks are processed
```

Changes:

```
-(EventProcessor) internal response event will not be dispatched to task
- (IdentityModule) getECIDAsync to get the response in callbacks.
```